crates/wasi-provider/src/handle.rs

```diff
@@ -0,0 +1,70 @@
+use std::io::SeekFrom;
+
+use tokio::io::{AsyncReadExt, AsyncSeekExt, BufReader};
+use tokio::sync::watch::Receiver;
+use tokio::task::JoinHandle;
+
+use kubelet::ContainerStatus;
+
+/// Represents a handle to a running WASI instance. Right now, this is
+/// experimental and just for use with the [crate::WasiProvider]. If we like
+/// this pattern, we will expose it as part of the kubelet crate
+pub struct RuntimeHandle<R: AsyncReadExt + AsyncSeekExt + Unpin> {
+    output: BufReader<R>,
+    handle: JoinHandle<Result<(), failure::Error>>,
+    status_channel: Receiver<ContainerStatus>,
+}
+
+impl<R: AsyncReadExt + AsyncSeekExt + Unpin> RuntimeHandle<R> {
+    /// Create a new handle with the given reader for log output and a handle to
+    /// the running tokio task. The sender part of the channel should be given
+    /// to the running process and the receiver half passed to this constructor
+    /// to be used for reporting current status
+    pub fn new(
+        output: R,
+        handle: JoinHandle<Result<(), failure::Error>>,
+        status_channel: Receiver<ContainerStatus>,
+    ) -> Self {
+        RuntimeHandle {
+            output: BufReader::new(output),
+            handle,
+            status_channel,
+        }
+    }
+
+    /// Write all of the output from the running process into the given buffer.
+    /// Returns the number of bytes written to the buffer
+    pub async fn output(&mut self, buf: &mut Vec<u8>) -> Result<usize, failure::Error> {
+        let bytes_written = self.output.read_to_end(buf).await?;
+        // Reset the seek location for the next call to read from the file
+        // NOTE: This is a little janky, but the Tokio BufReader does not
```

Comment:
I still don't think this is a helpful comment. If you feel like we should keep it, can you replace "This is a little janky, but" with "We must call `get_mut()` because"